src/imp/linux_raw/arch/outline/arm.S

```diff
@@ -0,0 +1,92 @@
+// arm syscall argument register ordering is the similar to the arm
+// userspace argument register ordering except that the syscall number
+// (nr) is passed in r7.
+//
+// arm.rs takes care of reordering the nr argument to the end for us,
+// so we only need to take care of sp by moving it to r4, r5 and r7.
+
+    .section    .text.rsix_reordered_syscall,"ax",%progbits
+    .p2align    4
+rsix_reordered_syscall:
+rsix_reordered_syscall0:
+rsix_reordered_syscall1:
+rsix_reordered_syscall2:
+rsix_reordered_syscall3:
+rsix_reordered_syscall4:
+rsix_reordered_syscall5:
+rsix_reordered_syscall6:
+    .cfi_startproc
```

Comment:
It looks like `clang` uses `.fnstart` and `.fnend` instead of `.cfi_startproc`/`.cfi_endproc` on ARM. ([AS docs](https://sourceware.org/binutils/docs/as/ARM-Directives.html#ARM-Directives).) Do you know if that'd be appropriate here?

It also adds `.cantunwind` to simple functions; syscalls shouldn't unwind, so should we add that here too?

Reply:
I used the same structure as the other architecture, so didn't pay to much attention about `.cfi_startproc` and `.cfi_endproc` but it indeed seems that it will be better to use `.fnstart` and `.fnend` (I have changed this). GLibc also use [`.fnstart`](https://github.com/bminor/glibc/blob/abd383584b16dd0fb1bbf40e4ece65ebe7b839ec/sysdeps/unix/sysv/linux/arm/sysdep.h#L226) and [`.fnend`](https://github.com/bminor/glibc/blob/abd383584b16dd0fb1bbf40e4ece65ebe7b839ec/sysdeps/unix/sysv/linux/arm/sysdep.h#L309).

As for `.cantunwind` I also added it because as you said syscalls shouldn't (can't ?) unwind. GLibc doens't use it for `syscall` but does for some other (a bit weird).